### ISSUE TITLE
How to handle "Back" button functionality

### ISSUE THREAD

**Issue by emilehay** [NoriginMedia/Norigin-Spatial-Navigation#57]
We've recently started using Norigin Spatial Navigation on our React-based Smart TV application and we love it! However, one integral part of navigating smart TV apps with the remote is the option to press the "back" button. I didn't see it in the documentation, nor do I see many people talking about it. Now I'm questioning my sanity for thinking this should be a feature of the library.

Anyways, how would you suggest we go about implementing this on our end? Would the best solution be a global listener for the back keycode and then setting the focus manually based on that? Any help would be appreciated, thanks!

**Comment by lcaprini** [comment]
Hi @emilehay, I had the same situation and yesterday I submitted a pull request that allows to handle the remote back button. The [pull request](https://github.com/NoriginMedia/Norigin-Spatial-Navigation/pull/56) is waiting for review, but in the meanwhile you could test it on my repo https://github.com/lcaprini/Norigin-Spatial-Navigation/tree/backButtonHandler.
Let me know if you found some bugs/issues.

--L

**Comment by emilehay** [comment]
Hi @lcaprini,

Wow! What a coincidence and thanks for the swift reply. I'll test it out your repo and let you know. Thanks!

**Comment by siarheipashkevich** [comment]
@lcaprini @emilehay all tv developers have the same issues )) thanks for the back feature and I've a question about `scroll`, do you implement own logic or use any library?

**Comment by lcaprini** [comment]
> about `scroll`, do you implement own logic or use any library?

I implemented my own logic based on CSS translate instead using the scroll API, because of some old TV devices don't support the "smooth" behavior on `scrollTo` API. I'm using  the info of focused `FocusableComponentLayout` and the position of the container.

**Comment by siarheipashkevich** [comment]
@lcaprini do you use `onFocus` callback and get the position of the focused element and manually scroll to this position?

**Comment by lcaprini** [comment]
Yes. I uses the example in this repo (src/App.tsx) as starting point, and customized it for my app.

**Comment by MacgyverMartins** [comment]
@emilehay @lcaprini are you developing for a specific TV model, like Samsung or LG? 
Each platform has its own key codes for the Back button. e.g Samsung is 10009. 
So what I did was use an event listener to handle the keyDown only when that app component is focused. Something like: 
```js

  const callbackHandler = useCallback((event) => {
    event.stopPropagation()
    if (event.keyCode === 10009) {
      // do something...
    }
  }, [])

  useEffect(() => {
    if (focused) window.addEventListener('keydown', callbackHandler);
    return () => {
      window.removeEventListener('keydown', callbackHandler);
    };
  }, [callbackHandler, register, targetEvent]);

**Comment by asgvard** [comment]
Hello,

We had some internal discussions and it seems like the consensus is that this library should not have a handling of a Back button logic. Back button is mostly related to the page navigation, and rarely to some UI interactions (like "close player controls"), but it has nothing to do with the *spatial/directional* navigation.
While we understand that for many it seems logical to compare it with the existing `onEnterPress`, I'm not sure I agree with this. When handling `onEnterPress` you are particularly interested in the element that is now focused and is sending this Enter press event. I.e. a Button telling you "I was pressed while being focused". So it makes more sense to have this event attached to every focusable element. Another reason is just historical, we added this callback long time ago and using it everywhere.
On the other hand, `onBackPress` is not related to a focused element (in most of the cases). Usually it's used for handling back presses to navigate to the previous page, and in this case you are not particularly interested in what was focused at this point, so adding such a listener to a library would not make sense.
It is straightforward to handle it by adding simple key listener (whether it's global or a local hook). In our projects we normally make a hook like `useBackHandler(callback)` that creates/destroys the key listener in the component you want to handle back press.
If we start adding more and more key handlers, it will bloat the library, so we'd rather keep it simple. Yes, `onEnterPress` is kind of an exception, while also being linked to a "currently focused element being pressed", so we thought it is not the same as other keys.

**Comment by jdunham2** [comment]
We are building an app where depending on where the user is when they press back, we want different behavior. We'd like to do something like this:

```
<Tablist onBackPress={focusMainNav />
<TabContent onBackPress={focusTabList} />
```
